### PR TITLE
#822: added length limit to cwd path in order to make the path more r…

### LINKED ISSUE #822: Handle long CWD paths in shell prompt
Inside the IDEasy shell, the CWD is displayed as part of the shell prompt, helping users to understand where they are within the file system. However, if the CWD path is very long (e.g., a deep folder structure), it can cause several problems:

Line Wrapping: The prompt may extend beyond the width of the terminal window, causing the path to wrap onto the next line, which can be visually unappealing and confusing for users.
Cluttered Display: Long paths can make the prompt hard to read and clutter the interface.
image (2)

In order to fix that, we should improve the shell prompt display by either truncating the CWD path or limiting its length to avoid long paths causing line wrapping in the terminal

### PR DESCRIPTION
**Pull request by marceltchanga9**
### Implemented changes:

 Added line separator when the cwd path is longer than 80 
---

### Testing instructions

1. Create in the project folder a  sub-folder with a name having more than 80 characters ( eg : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaatestlength)
2. Run the CLI Application with argument --shell
3. Navigate to the folder you created in the command line ( cd ...)
4. Check that the line breaks and "$ide" is at the beginning of a new line
5. To make sure that the test works, you can also test it with folder's name shorter or longer than 80
6. Delete the folder(s) you created
---

### Checklist for this PR

Make sure everything is checked before merging this PR. For further info please also see
our [DoD](https://github.com/devonfw/IDEasy/blob/main/documentation/contributing/DoD.adoc).

- [x] When running `mvn clean test` locally all tests pass and build is successful
- [x] PR title is of the form `#«issue-id»: «brief summary»` (e.g. `#921: fixed setup.bat`). If no issue ID exists, title only.
- [x] PR top-level comment summarizes what has been done and contains link to addressed issue(s)
- [x] PR and issue(s) have suitable labels
- [x] Issue is set to `In Progress` and assigned to you *or* there is no issue (might happen for very small PRs)
- [x] You followed all [coding conventions](https://github.com/devonfw/IDEasy/blob/main/documentation/coding-conventions.adoc)
- [x] You have added the issue implemented by your PR in [CHANGELOG.adoc](https://github.com/devonfw/IDEasy/blob/main/CHANGELOG.adoc) unless issue is labeled
  with `internal`
- [x] You have formulated clear instructions on how to test your contribution under "Testing instructions"

### CODE REVIEW

**Comment by oanding-blrng** [comment]
I'm only able to recreate the original problem mentioned in #822. It is not clear to me, how to properly test the correct behavior implemented in this ticket/pull request.

@marceltchanga9 Please provide clearer steps for testing and add appropriate labels, which match the labels of the original ticket.

**Comment by marceltchanga9** [comment]
> I'm only able to recreate the original problem mentioned in #822. It is not clear to me, how to properly test the correct behavior implemented in this ticket/pull request.
> 
> @marceltchanga9 Please provide clearer steps for testing and add appropriate labels, which match the labels of the original ticket.

@oanding-blrng I updated the test steps. They should be now more understandable.

**Comment by oanding-blrng** [comment]
@marceltchanga9 Thank you for the test clarifications. I can confirm, that with a certain length of the cwd, an automatic line break will be added. This should indeed make working in really nested directories more easily.

Still there are two things to do:
- sign the CLA 
- Merge the latest changes from upstream
- update the `changelog.adoc`, because the target release for this pr changed

`mvn test` also runs successfully. After the sign of the CLA and the update to the `changelog.adoc`, this pr can move foreward.

**Comment by oanding-blrng** [comment]
The points mentioned in the latest comment are has been addressed, this pr can move on.

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -6,13 +6,13 @@ This file documents all notable changes to https://github.com/devonfw/IDEasy[IDE
...
 Release with new features and bugfixes:
 
+* https://github.com/devonfw/IDEasy/issues/822[#822] : Added Cwd path limit in shell
 
 The full list of changes for this release can be found in https://github.com/devonfw/IDEasy/milestone/46?closed=1[milestone 2026.07.001].
 
 == 2026.06.001
 
 Release with new features and bugfixes:
-
```
Comment (hohwille) [inline CHANGELOG.adoc:15]: You accidentally removed this line from the previous release.
Please revert that change.

**Review by hohwille — APPROVED** [review]
@marceltchanga9 thanks for your PR. Your change is fine and ready for merge 👍 
When adding a new feature it is however, best-practice to also add a JUnit test to cover the feature. Since this shell is rather an edge-case and more tricky to test, I will approve and merge as is.

File: CHANGELOG.adoc (hunk 1 above)
Comment (hohwille) [reply CHANGELOG.adoc:15]: To be pragmatic, I will change it for you, but please notice and take care for next PRs.

**Comment by hohwille** [comment]
Rested successfully:
```
D:\projects\IDEasy\workspaces\main\IDEasy\cli\src\main\java\com\devonfw\tools\ide\validation\aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
$ ide version
2026.07.001-07_01_03-SNAPSHOT
D:\projects\IDEasy\workspaces\main\IDEasy\cli\src\main\java\com\devonfw\tools\ide\validation\aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
$ ide
```
